src/pkg/packager/sources/utils.go

```diff
@@ -36,46 +37,72 @@ func IsValidFileExtension(filename string) bool {
 	return false
 }
 
+// identifyUnknownTarball tries "path" as-is first, then retries
+// with .tar.zst, .tar.gz, .tar.xz, and .tar appended,
+// using archives.Identify to detect only tar variants.
 func identifyUnknownTarball(path string) (string, error) {
+	// 1) missing file?
 	if helpers.InvalidPath(path) {
 		return "", &os.PathError{Op: "open", Path: path, Err: os.ErrNotExist}
 	}
-	if filepath.Ext(path) != "" && IsValidFileExtension(path) {
-		return path, nil
-	} else if filepath.Ext(path) != "" && !IsValidFileExtension(path) {
-		return "", fmt.Errorf("%s is not a supported tarball format (%+v)", path, GetValidPackageExtensions())
-	}
+	ctx := context.Background()
 
-	// rename to .tar.zst and check if it's a valid tar.zst
-	tzst := fmt.Sprintf("%s.tar.zst", path)
-	if err := os.Rename(path, tzst); err != nil {
-		return "", err
-	}
-	// TODO(mkcp): See https://github.com/zarf-dev/zarf/issues/3051
-	format, err := archiver.ByExtension(tzst)
-	if err != nil {
-		return "", err
-	}
-	_, ok := format.(*archiver.TarZstd)
-	if ok {
-		return tzst, nil
-	}
+	// helper to test a candidate filename
+	try := func(name string) (bool, error) {
+		f, err := os.Open(name)
+		if err != nil {
+			return false, err
+		}
+		defer f.Close()
 
-	// rename to .tar and check if it's a valid tar
-	tb := fmt.Sprintf("%s.tar", path)
-	if err := os.Rename(tzst, tb); err != nil {
-		return "", err
+		// Identify by filename or header
+		format, _, err := archives.Identify(ctx, filepath.Base(name), f)
+		if err != nil {
+			// NoMatch or other error
+			return false, nil
+		}
+
+		// "format" might be a plain Tar, or a CompressedArchive wrapping Tar
+		switch v := format.(type) {
+		case archives.Tar:
+			return true, nil
+		case archives.CompressedArchive:
+			if _, ok := v.Archival.(archives.Tar); ok {
+				return true, nil
+			}
+		}
+		return false, nil
 	}
-	format, err = archiver.ByExtension(tb)
-	if err != nil {
+
+	// 2) try original path
+	if _, err := try(path); err != nil {
 		return "", err
 	}
-	_, ok = format.(*archiver.Tar)
-	if ok {
-		return tb, nil
+	// else if ok {
+	// 	return path, nil
+	// }
+
+	// 3) try each extension in order
+	for _, ext := range []string{".tar.zst", ".tar.gz", ".tar.xz", ".tar"} {
```

Comment:
Similar magic-string comment as in `archive` but in this case it makes sense to me to keep this string range local to the function `identifyUnknownTarball`.